### PR TITLE
Fix migrations

### PR DESCRIPTION
**Pull request by janherich**
[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)

addresses #2488 

Fixes migrations for previously falling corner cases (`/request` commands)

Testing notes: Test upgrade path from 0.9.12

status: ready

### CODE REVIEW

**Comment by oskarth** [comment]
> Testing notes: Test upgrade path from 0.9.12

^ please add this to description if it is accurate @janherich

File: src/status_im/data_store/realm/schemas/account/v19/core.cljs [hunk 1]
```diff
@@ -89,4 +95,5 @@
   (remove-contact! new-realm "transactor-personal")
   (remove-contact! new-realm "transactor-group")
   (update-commands (juxt :bot :command) owner-command->new-props new-realm "command")
-  (update-commands (juxt :command) console-requests->new-props new-realm "command-request"))
+  (update-commands (juxt :command) console-requests->new-props new-realm "command-request")
+  (update-commands (juxt :command (comp count :prefill)) transactor-requests->new-props new-realm "command-request"))
```
Comment (oskarth) [inline src/status_im/data_store/realm/schemas/account/v19/core.cljs:99]: Can you explain why this change is necessary?

File: src/status_im/data_store/realm/schemas/account/v19/core.cljs [hunk 2]
```diff
@@ -73,6 +73,12 @@
    ["phone"] {:content-command-ref ["console" :response 50 "phone"]}
    ["confirmation-code"] {:content-command-ref ["console" :response 50 "confirmation-code"]}})
 
+(def transactor-requests->new-props
+  {;; former transactor-personal request
+   ["send" 1] {:content-command-ref ["transactor" :response 83 "send"]}
+   ;; former transactor-group request
+   ["send" 2] {:content-command-ref ["transactor" :response 85 "send"]}})
```
Comment (oskarth) [inline src/status_im/data_store/realm/schemas/account/v19/core.cljs:80]: What's magical about 83 and 85?

**Comment by janherich** [comment]
@oskarth done

**Comment by oskarth** [comment]
> "/send", "request", "/debug" etc. commands are displayed without icons in chat (all chat types).

Issue says all these, PR says  `(/request commands)` - why discrepancy?
EDIT: This is clarified in issue thread.

File: src/status_im/data_store/realm/schemas/account/v19/core.cljs (hunk 1 above)
Comment (janherich) [reply src/status_im/data_store/realm/schemas/account/v19/core.cljs:99]: Because I completely forgot that `/request` commands are creating messages of the type "command-request", so we have to migrate them as well.

Comment (oskarth) [reply src/status_im/data_store/realm/schemas/account/v19/core.cljs:99]: sorry I mean, what is for example `prefill` about? lacking some context

File: src/status_im/data_store/realm/schemas/account/v19/core.cljs (hunk 2 above)
Comment (janherich) [reply src/status_im/data_store/realm/schemas/account/v19/core.cljs:80]: This is bitmask (as a decimal number) computed from command scope array on js side.
I'm not super thrilled about bitmasking as well (it's not like we are coding software for 8bit micro-controller and need to save every bit of memory), but as the javascript is unfortunately not supporting sets as a first class collection, we have to somehow encode (scopes) array in a way which produces constant result independent of the order of elements in it (or duplicates).

**Comment by janherich** [comment]
@oskarth please read the Issue carefully and scroll to the very end for the latest test results described by @annadanchenko

File: src/status_im/data_store/realm/schemas/account/v19/core.cljs (hunk 2 above)
Comment (oskarth) [reply src/status_im/data_store/realm/schemas/account/v19/core.cljs:80]: Oh I see, thanks for the explanation. I realize this is out of scope of PR to improve this, but do you know if there is an issue to improve this? I'm certain we can come up with some better than decimal bitmasks to enode scopes.

**Review by oskarth — APPROVED** [review]
Lacking some context to fully understand necessity of changes, but as long we as test upgrade path that seems OK by me.

File: src/status_im/data_store/realm/schemas/account/v19/core.cljs (hunk 1 above)
Comment (oskarth) [reply src/status_im/data_store/realm/schemas/account/v19/core.cljs:99]: So I would naively imagine prefill being a fn of whatever command is and not a standalone thing, but that evidently doesn't appear to be the case.

Comment (janherich) [reply src/status_im/data_store/realm/schemas/account/v19/core.cljs:99]: `:prefill` is array of prefilled arguments, we distinguish between group command/personal command based on it's size 
It's cryptic as sh*t, but it's only migration code, never bothering anybody once we move past v18 -> v19 migration

**Review by goranjovic — APPROVED** [review]
